GameServerLib/GameObjects/AttackableUnits/AttackableUnit.cs

```diff
@@ -170,14 +190,24 @@ public bool ChangeModel(string model)
                 //Damage dealing. (based on leagueoflegends' wikia)
                 damage = defense >= 0 ? 100 / (100 + defense) * damage : (2 - 100 / (100 - defense)) * damage;
             }
-
             ApiEventManager.OnUnitDamageTaken.Publish(this);
 
             Stats.CurrentHealth = Math.Max(0.0f, Stats.CurrentHealth - damage);
             if (!IsDead && Stats.CurrentHealth <= 0)
             {
                 IsDead = true;
                 Die(attacker);
+                if (useShield)
+                {
+                    for (ShieldType shield = 0; shield < ShieldType.END;shield++)
+                    {
+                        if (ShieldAmount[(int)shield] > 0)

```

Comment:
This line make codacy unnecessarily upset. You can fix it by putting `ShieldAmount[(int)shield]` into its own variable.

Reply:
no, i just make `if(useShield)` into `if(ShouldAmount[(int)shield] >0)`
Codacy seems good now